pkg/kyc/component.go

```diff
@@ -0,0 +1,212 @@
+package kyc
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	"github.com/cloudtrust/keycloak-client"
+
+	cs "github.com/cloudtrust/common-service"
+	"github.com/cloudtrust/common-service/database"
+	errorhandler "github.com/cloudtrust/common-service/errors"
+	apikyc "github.com/cloudtrust/keycloak-bridge/api/kyc"
+	"github.com/cloudtrust/keycloak-bridge/internal/dto"
+	"github.com/cloudtrust/keycloak-bridge/internal/keycloakb"
+	internal "github.com/cloudtrust/keycloak-bridge/internal/keycloakb"
+	messages "github.com/cloudtrust/keycloak-bridge/internal/messages"
+	kc "github.com/cloudtrust/keycloak-client"
+)
+
+// KeycloakClient are methods from keycloak-client used by this component
+type KeycloakClient interface {
+	UpdateUser(accessToken string, realmName, userID string, user kc.UserRepresentation) error
+	GetUsers(accessToken string, reqRealmName, targetRealmName string, paramKV ...string) (kc.UsersPageRepresentation, error)
+}
+
+// ConfigurationDBModule is the interface of the configuration module.
+type ConfigurationDBModule interface {
+	GetConfiguration(context.Context, string) (dto.RealmConfiguration, error)
+}
+
+// Component is the register component interface.
+type Component interface {
+	GetActions(ctx context.Context) ([]apikyc.ActionRepresentation, error)
+	GetUser(ctx context.Context, socialRealmName string, username string) (apikyc.UserRepresentation, error)
+	ValidateUser(ctx context.Context, socialRealmName string, username string, user apikyc.UserRepresentation) error
+}
+
+// Component is the management component.
+type component struct {
+	keycloakClient KeycloakClient
+	tokenProvider  keycloak.OidcTokenProvider
+	usersDBModule  keycloakb.UsersDBModule
+	configDBModule ConfigurationDBModule
+	eventsDBModule database.EventsDBModule
+	logger         internal.Logger
+}
+
+// NewComponent returns the management component.
+func NewComponent(keycloakClient KeycloakClient, tokenProvider keycloak.OidcTokenProvider, usersDBModule keycloakb.UsersDBModule, configDBModule ConfigurationDBModule, eventsDBModule database.EventsDBModule, logger internal.Logger) Component {
+	return &component{
+		keycloakClient: keycloakClient,
+		tokenProvider:  tokenProvider,
```

Comment:
For KYC, I think tokenProvider should not be needeed as the user must has rights to perform the action

Reply:
Forgot to remove that when splitting previous version